Gems/Camera/Code/Source/CameraSystemComponent.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "CameraSystemComponent.h"
+
+#include <AzCore/Math/MatrixUtils.h>
+#include <AzCore/Component/TransformBus.h>
+
+#include <AzFramework/Viewport/CameraState.h>
+
+#include <Atom/RPI.Public/View.h>
+#include <Atom/RPI.Public/ViewportContextBus.h>
+
+namespace Camera
+{
+    void CameraSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<CameraSystemComponent, AZ::Component>()
+                ->Version(1)
+                ;
+        }
+    }
+
+    void CameraSystemComponent::Activate()
+    {
+        CameraSystemRequestBus::Handler::BusConnect();
+        ActiveCameraRequestBus::Handler::BusConnect();
+        CameraNotificationBus::Handler::BusConnect();
+    }
+
+    void CameraSystemComponent::Deactivate()
+    {
+        CameraSystemRequestBus::Handler::BusDisconnect();
+        ActiveCameraRequestBus::Handler::BusDisconnect();
+        CameraNotificationBus::Handler::BusDisconnect();
```

Comment:
uber nit: This probably doesn't make any different here but in general we try and disconnect in the reverse order to the order we connect for buses (just to avoid any weird ordering issues that might crop up) So these could be reordered if you would like 🙂